contracts/YearnV2Strat.sol

```diff
@@ -0,0 +1,130 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.7.3;
+
+import "./IStrat.sol";
+import "./IYToken.sol";
+import "./IVault.sol";
+import "./Timelock.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+contract YTokenStrat is IStrat {
+    
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    using SafeERC20 for IERC20Detailed;
+    IVault public vault;
+    IYToken public yToken;
+    IERC20Detailed public underlying;
+    Timelock public timelock;
+    uint public minWithdrawalCap; // prevents the owner from completely blocking withdrawals
+    uint public withdrawalCap = uint(-1); // max uint
+    uint public buffer; // buffer of underlying to keep in the strat
+    string public name = "Yearn V2"; // for display purposes only
+
+    modifier onlyVault {
+        require(msg.sender == address(vault));
+        _;
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == vault.owner()); // vault owner is strat owner
+        _;
+    }
+
+    constructor(IVault vault_, IYToken yToken_) {
+        vault = vault_;
+        yToken = yToken_;
+        timelock = Timelock(vault.timelock()); // use the same timelock from the vault
+        underlying = IERC20Detailed(yToken_.token());
+        underlying.safeApprove(address(yToken), uint(-1)); // intentional underflow
+        minWithdrawalCap = 1000 * (10 ** underlying.decimals()); // 10k min withdrawal cap
+    }
+
+    function invest() external override onlyVault {
+        uint balance = underlying.balanceOf(address(this));
+        if(balance > buffer) {
+            uint max = yToken.availableDepositLimit();
+            yToken.deposit(Math.min(balance - buffer, max)); // can't underflow because of above if statement
+        }
+    }
+
+    function divest(uint amount) external override onlyVault {
+        uint balance = underlying.balanceOf(address(this));
+        if(balance < amount) {
+            uint missingAmount = amount - balance; // can't underflow because of above it statement
+            require(missingAmount <= withdrawalCap, "Reached withdrawal cap"); // Big withdrawals can cause slippage on Yearn's side. Users must split into multiple txs
+            yToken.withdraw(sharesForAmount(missingAmount) + 1);  // +1 is a fix for a rounding issue
```

Comment:
i'd add do a Math.min(sharesForAmount(missingAmount)+1, ytoken.balanceOf(this)) here

Reply:
ok